plenum/server/view_change/view_changer.py

```diff
@@ -521,6 +525,9 @@ def startViewChange(self, proposed_view_no: int):
         # TODO: view change is a special case, which can have different
         # implementations - we need to make this logic pluggable
 
+        if self.pre_vc_strategy and not continue_vc:
```

Comment:
I think we should no call the strategy if `propagate_primary is True` (please also add tests for this)